mux.go

```diff
@@ -0,0 +1,215 @@
+		panic(err)
+	}
+}
+
+// addHandlerSafely adds the handler for the given entrypoint to the mux.
+// If there's an error, usually by registering the same entrypoint twice, an error is returned.
+func addHandlerSafely[T, TContext Data](m *mux, entrypoint string, handler Handler[T, TContext]) error {
+	_, exists := m.handlers[entrypoint]
+	if exists {
+		return fmt.Errorf("entrypoint %q is already registered on this mux", entrypoint)
+	}
+
+	m.handlers[entrypoint] = &typedHandlerFactory[T, TContext]{h: handler}
+	return nil
+}
+
+// parseArguments takes a list of arguments, usually [os.Args] and parses them into a valid [options].
+//
+// # Rules
+//
+// The following rules apply:
+//
+//   - args must contain either two or five elements.
+//   - If it contains two elements, the second element must be `metadata` in order for [options.WriteMetadata] to be true.
+//
+// # Argument order
+//
+// Otherwise, the arguments need to be in the following order:
+//
+//  1. The name of the binary.
+//  2. The entrypoint that should be called, used for determining the [Handler].
+//  3. The standard output termination sequence, which is written *before* writing the serialized JSON response.
+//  4. Whether the daemon should be kept alive, determined by a 0 (false) or a 1 (true).
+//  5. The daemon execution termination sequence, which is written *after* writing the *successful* JSON response.
+func parseArguments(args []string) (options, error) {
+	if len(args) == 2 && args[1] == "metadata" {
+		return options{WriteMetadata: true}, nil
+	}
+
+	// Check number of arguments:
+	// binary name, entrypoint, os.Stdout termination, keepalive enabled, daemon execution termination
+	if argCount := len(args); argCount != 5 {
+		return options{}, fmt.Errorf("invalid number of process arguments: %d", argCount)
+	}
+
+	res := options{
+		Entrypoint:                         args[1],
+		StdoutExecutionSequence:            strings.ReplaceAll(args[2], "\\0", "\x00"),
+		KeepAlive:                          args[3] == "1",
+		DaemonExecutionTerminationSequence: strings.ReplaceAll(args[4], "\\0", "\x00"),
+	}
+	return res, nil
+}
+
+// Start starts the mux and returns potential runtime errors.
+//
+// The startup arguments are read from [os.Args], however it's possible to set custom startup options
+// by setting the options on this mux.
+//
+// If [options.WriteMetadata] is true, JSON-formatted metadata is written to [os.Stdout]
+// and the method returns.
+//
+// If [options.KeepAlive] is true, the goroutine is blocked and can be cancelled
+// via the context. It also listens for incoming [syscall.SIGINT] signals and stops gracefully.
+//
+// # Example usage
+//
+//	mux := e5e.NewMux()
+//	log.Fatalf("mux error: %v", mux.Start(context.Background()))
+func (m *mux) Start(ctx context.Context, opts options) error {
+	if opts.WriteMetadata {
+		writeMetadata()
+		return nil
+	}
+
+	scan := bufio.NewScanner(os.Stdin)
+	scan.Buffer([]byte{}, 4<<10)
```

Comment:
This limit is way too low. We need to be able to process uploads with mutliple MBs. 
Additionally, I don't like this limit being enforced in the library code. This is something that e5e should take care of. The library should assume that the stdin is trusted. Therefore, I'd suggest using the original `ReadLine`.